[4.x] Add "All Getters" for "Flash" values and "Add"

jakejohns opened the pull request:
This is just #47 for 3.x. 
It looks like there was already a `getAll` equivalent: `getSegment`.

jakejohns committed:
Fx README syntax

jakejohns committed:
Add Flash "Get All" methods

- Segment::getFlashAll();
- Segment::getFlashNextAll();

harikt commented on src/Segment.php:
```diff
@@ -167,6 +167,23 @@ public function getFlash($key, $alt = null)
+    public function getFlashAll($alt = array())
+    {
+        $this->resumeSession();
+        return isset($_SESSION[Session::FLASH_NOW][$this->name])
```
Doesn't flash all means flash now + next ?

jakejohns commented:
Fair point, @harikt. Perhaps the name is a little terse.
Perhaps `getAllCurrentFlash()`, and `getAllNextFlash()` would be better?

harikt commented:
Let's wait to hear  from pmjones thoughts.

jakejohns committed:
Change 'getAll' flash methods to less terse naming

jakejohns committed:
Add `add` methods to append numeric key values

jakejohns commented:
I added "add" methods here as well.
I should have perhaps done this as a separate PR, but the intention is that the `getAll` and `add` stuff would be complimentary. 

@harikt, I just noticed that you said to wait on pmj, but i'm pretty sure you're right, so I had already gone and changed it. Obviously can always change to something else.

As an illustration of my intended use case:

``` php
<?php
// @codingStandardsIgnoreFile

abstract class MyAbstractResponder
{
    protected $msgSegment = 'My\\Messages';

    protected function addMessage($msg, $level = 'info')
    {
        $this->getSession()
            ->getSegment($this->msgSegment)
            ->addFlash(['lvl' => $level, 'msg' => $msg]);
    }

    protected function getMessages()
    {
        return $this->getSession()
            ->getSegment($this->msgSegment)
            ->getAllCurrentFlash([]);
    }

    protected function render()
    {
        $view = $this->getView();
        $view->addData(['messages' => $this->getMessages()]);
        $this->response->getBody()->write($view());
    }

    protected function updated()
    {
        $this->addMessage('Successfully updated thing', 'success');
        $this->redirect();
    }

    //...
}

```

harikt commented:
@jakejohns Sorry that this is not merged. I am going to release a 4.x for session to make it compatible with 8.x . Newer feature is not incorported for now. So this is currently put on hold.
